io-engine/src/rebuild/rebuild_descriptor.rs

```diff
@@ -183,4 +185,83 @@ impl RebuildDescriptor {
                 bdev: self.dst_uri.clone(),
             })
     }
+
+    /// Verify segment copy operation by reading destination, and comparing with
+    /// the source.
+    pub(super) async fn verify_segment(
+        &self,
+        offset_blk: u64,
+        iovs: &mut [IoVec],
+    ) -> Result<(), RebuildError> {
+        // Read the source again.
```

Comment:
Why do we need to read source again? The LBA ranges are still locked I believe when `verify_segment` is called and the buffers that we used to write to destination can be still valid.

Reply:
There can be source read errors.

Reply:
hmm, not sure if that should matter here. IIUC, the intent is to read from destination, and compare to the buffer from which we wrote.